nestle_core/src/lib.rs

```diff
@@ -105,3 +121,44 @@ impl_for_primitive!(i8, u8, 8);
 impl_for_primitive!(i16, u16, 16);
 impl_for_primitive!(i32, u32, 32);
 impl_for_primitive!(i64, u64, 64);
+impl_for_primitive!(u8, u8, 8);
+impl_for_primitive!(u16, u16, 16);
+impl_for_primitive!(u32, u32, 32);
+impl_for_primitive!(u64, u64, 64);
+impl_for_primitive_with_cast!(
+    f32,
+    |this: f32| { this.to_bits() as i64 },
+    |value: i64| { Some(f32::from_bits(value as u32)) },
+    32
+);
+impl_for_primitive_with_cast!(
+    f64,
+    |this: f64| { this.to_bits() as i64 },
+    |value: i64| { Some(f64::from_bits(value as u64)) },
+    64
+);
+impl_for_primitive_with_cast!(
+    bool,
+    |this: bool| { this as i64 },
+    |value: i64| { Some(value != 0) },
+    8
+);
+impl_for_primitive_with_cast!(
+    char,
+    |this: char| { this as i64 },
+    |value: i64| { char::from_u32(value as u32) },
+    32
+);
+
+impl Nestle for () {
```

Comment:
Hmmm... Nice idea but I think doing this for ZSTs introduces for trouble than it's worth. I can't think of the use-case.

Reply:
There is a use case in `DiverseZoo`, `Empty` is that case. We can't just write `Empty = -112` instead(`Tuple-style data must be provided for all choices (tick "Empty").`), but we often have circumstances where we need a case that doesn't have values among cases with values.

Reply:
Hmm, maybe we can use `pub struct Empty {}`
instead, but I don't think the unit will bring many problems

Reply:
In general, I think that "unit" will never be superfluous. But if you still think that it is redundant, then I will remove it.

Reply:
You should be able to express that with just a regular enum variant Empty rather than Empty(Empty).

Yeah I think it would be best to omit it for now - I think it has the potential to be misleading.

Reply:
It turns out the macro expects you to either use tuple style enums or regular, and doesn't let you mix both. Seeing as that's the case we can just keep this as it is and address later.